src/components/Modals/QrCode/Form.tsx

```diff
@@ -97,27 +98,52 @@ export const Form: React.FC<QrCodeFormProps> = ({ accountId }) => {
     (decodedText: string) => {
       ;(async () => {
         try {
-          console.log('QR decode text:', decodedText)
           // If this is a WalletConnect dApp QR Code, skip the whole send logic and render the QR Code Modal instead.
           // There's no need for any RFC-3986 decoding here since we don't really care about parsing and WC will do that for us
           if (decodedText.startsWith('wc:')) return setWalletConnectDappUrl(decodedText)
 
-          // This should
-          // - Parse the address, amount and asset. This should also exhaust URI parsers (EVM and UTXO currently) and set the amount/asset if applicable
-          // - If there is a valid asset (i.e UTXO, or ETH, but not ERC-20s because they're unsafe), populates the asset and goes directly to the address step
-          // If no valid asset is found, it should go to the select asset step
-          const maybeUrlResult = await parseMaybeUrl({ urlOrAddress: decodedText })
+          // Try parsing as payment URI first, otherwise use legacy address parser
+          const urlDirectResult = parseUrlDirect(decodedText)
```

Comment:
`parseUrlDirect()` does precisely that - attempts parsing an URL.
It's now a single concern utility that does precisely that, with all the heavy URL logic moved out of the previous utils.